docs/contribute/dev-setup.md

```diff
@@ -0,0 +1,112 @@
+
+## Initial setup
+
+The following command downloads and builds the most recent development version of Storybook on your local machine:
+
+```sh
+git clone https://github.com/storybookjs/storybook.git
+cd storybook
+yarn bootstrap --core
+```
+
+> **_Note:_** On Windows, you may need to run `yarn` before `yarn bootstrap`!
+
+## Sanity check
+
+After [the initial setup](#initial-setup) you should have a fully built version of Storybook on your local machine. At this point, i's useful to verify that things are working, both with automated tests and by hand.
+
+First, run the unit tests:
+
+```sh
+yarn test
+```
+
+Then, run an example project:
+
+```
+cd examples/cra-ts-essentials
+yarn storybook
+```
+
+There are a bunch of example projects corresponding to different frameworks and app environments. They automatically run against the code in the monorepo.
+
+## Interactive development
+
+Now that you've [verified your setup](#sanity-check), it's time to jump into code. The easiest way to do this is to have one of the examples running in one terminal window, and to have an interactive build process running in another terminal.
+
+Assuming, you're still running `examples/cra-ts-essentials` from the previous step, open a new terminal and run the build process there from the root of the Storybook monorepo:
+
+```sh
+yarn build
+```
+
+Select the `watch` option to develop interactively. Then select one more libraries to build. For example, if you are developing a feature for `@storybook/addon-docs` you might want to select `@storybook/addon-docs` and `@storybook/components`.
+
+<details>
+
+<summary>`yarn dev` watches everything but is resource-intensive</summary>
+
+It's a pain to have to know which packages you're going to edit ahead of time, but watching the entire package structure is prohibitively expensive on most machines. If you're on a fast machine or simply enjoy hearing your CPU fan spin up like a jet engine, you can use `yarn dev` instead of `yarn build`.
+
+</details>
+
+Now, when you edit a file in one of the watched packages, it will automatically re-build.
+
+If the file affects the `preview` (the inner iframe of Storybook, where the stories contents live), the preview should automatically refresh a second or two after you save your changes.
+
+If the file affects the `manager` (the outer iframe where the UI and addon panels live), you need to manually refresh the browser.
+
+## Reproductions
+
+Since May 2021 we're strongly encouraging users to create reproductions for their issues. Just like it's possible to [interactively develop](#interactive-development) against example projects in the storybook monorepo, it's also possible to interactively develop against a reproduction repo.
+
+To do so, run the following command in the root of the Storybook monorepo:
+
+```sh
+npx sb@next link https://github.com/<some-user>/<some-project>.git
```

Comment:
@shilman does `sb@next` link also take a directory as a parameter? I.e, if you want to run the development Storybook against one of your own local Storybooks?

Reply:
Great question, currently it expects the input to be a repository url. I think the effort is not really big to add this feature though! It would be quite handy.

Reply:
I'll add it @zol

Reply:
Brilliant, ty @shilman

Reply:
https://github.com/storybookjs/storybook/pull/14950